api/config_entry_test.go

```diff
@@ -0,0 +1,146 @@
+		require.NotEqual(t, 0, qm.RequestTime)
+
+		// verify it
+		readProxy, ok := entry.(*ProxyConfigEntry)
+		require.True(t, ok)
+		require.Equal(t, global_proxy.Kind, readProxy.Kind)
+		require.Equal(t, global_proxy.Name, readProxy.Name)
+		require.Equal(t, global_proxy.Config, readProxy.Config)
+
+		// update it
+		global_proxy.Config["baz"] = true
+		wm, err = config_entries.Set(global_proxy, nil)
+		require.NoError(t, err)
+		require.NotNil(t, wm)
+		require.NotEqual(t, 0, wm.RequestTime)
+
+		// list it
+		entries, qm, err := config_entries.List(ProxyDefaults, nil)
+		require.NoError(t, err)
+		require.NotNil(t, qm)
+		require.NotEqual(t, 0, qm.RequestTime)
+		require.Len(t, entries, 1)
+		readProxy, ok = entries[0].(*ProxyConfigEntry)
+		require.True(t, ok)
+		require.Equal(t, global_proxy.Kind, readProxy.Kind)
+		require.Equal(t, global_proxy.Name, readProxy.Name)
+		require.Equal(t, global_proxy.Config, readProxy.Config)
+
+		// delete it
+		wm, err = config_entries.Delete(ProxyDefaults, ProxyConfigGlobal, nil)
+		require.NoError(t, err)
+		require.NotNil(t, wm)
+		require.NotEqual(t, 0, wm.RequestTime)
+
+		entry, qm, err = config_entries.Get(ProxyDefaults, ProxyConfigGlobal, nil)
+		require.Error(t, err)
+	})
+
+	t.Run("Service Defaults", func(t *testing.T) {
+		service := &ServiceConfigEntry{
+			Kind:     ServiceDefaults,
+			Name:     "foo",
+			Protocol: "udp",
+		}
+
+		service2 := &ServiceConfigEntry{
+			Kind:     ServiceDefaults,
+			Name:     "bar",
+			Protocol: "tcp",
+		}
+
+		// set it
+		wm, err := config_entries.Set(service, nil)
+		require.NoError(t, err)
+		require.NotNil(t, wm)
+		require.NotEqual(t, 0, wm.RequestTime)
+
+		// also set the second one
+		wm, err = config_entries.Set(service2, nil)
+		require.NoError(t, err)
+		require.NotNil(t, wm)
+		require.NotEqual(t, 0, wm.RequestTime)
+
+		// get it
+		entry, qm, err := config_entries.Get(ServiceDefaults, "foo", nil)
+		require.NoError(t, err)
+		require.NotNil(t, qm)
+		require.NotEqual(t, 0, qm.RequestTime)
+
+		// verify it
+		readService, ok := entry.(*ServiceConfigEntry)
+		require.True(t, ok)
+		require.Equal(t, service.Kind, readService.Kind)
+		require.Equal(t, service.Name, readService.Name)
+		require.Equal(t, service.Protocol, readService.Protocol)
+
+		// update it
+		service.Protocol = "tcp"
+		wm, err = config_entries.Set(service, nil)
+		require.NoError(t, err)
+		require.NotNil(t, wm)
+		require.NotEqual(t, 0, wm.RequestTime)
+
+		// list them
+		entries, qm, err := config_entries.List(ServiceDefaults, nil)
+		require.NoError(t, err)
+		require.NotNil(t, qm)
+		require.NotEqual(t, 0, qm.RequestTime)
+		require.Len(t, entries, 2)
+
+		for _, entry = range entries {
+			if entry.GetName() == "foo" {
+				readService, ok = entry.(*ServiceConfigEntry)
+				require.True(t, ok)
+				require.Equal(t, service.Kind, readService.Kind)
+				require.Equal(t, service.Name, readService.Name)
+				require.Equal(t, service.Protocol, readService.Protocol)
+			}
+		}
```

Comment:
Shouldn't we verify both here? service2/"bar" never gets checked.

Reply:
Maybe this deserves a comment in the code.

Basically we have the right number of entries and I am not trying to validate the correctness of all the entries. This validation is really to verify that the updated entry was updated properly.

Reply:
I updated the tests to verify both anyways.